### ISSUE TITLE
org-roam-insert can't handle accented characters

### ISSUE THREAD

**Issue by soldeace** [jethrokuan/org-roam#158]
Whenever I create a new file with `org-roam-insert`, accented letters are omitted from the created file (although the #+TITLE turns out okay). For example, if I try to create a file named "Sobre a formação dos planetas", the resulting filename is "202002212123_sobre_a_formao_dos_planetas".

**Comment by herbertjones** [comment]
If you want the filenames to match, you can modify `org-roam-file-name-function` and replace `org-roam--file-name-timestamp-title`.

Internally `org-roam--title-to-slug` is used, which could be improved to allow unicode alphanumeric characters.  In my personal config I have a function that is similar to what it does.  If there are no objections, perhaps it could be upgraded to better handle non-ascii alphanumerics.

```elisp
(defun org-roam--title-to-slug (name)
  "Convert filename into one without unacceptable characters or
spaces, replacing with dashes and underscores."
  (let ((pairs (list '("\\\\" . "_backslash_")   ;; single backslash
                     '("/" . "_slash_")
                     '("\\." . "_dot_")  ;; dot
                     '(" " . "_")
                     (cons (rx (not (any alnum digit))) "_")
                     '("__*" . "_") ;; no sequential underscores
                     '("_$" . ""))))
    (--reduce-from (s-downcase (replace-regexp-in-string (car it) (cdr it) acc)) name pairs)))

(org-roam--title-to-slug "Sobre a formação dos planetas")
"sobre_a_formação_dos_planetas"

(org-roam--title-to-slug "Sobre la formación de los planetas")
"sobre_la_formación_de_los_planetas"

(org-roam--title-to-slug "1\\2.3///**")
"1_backslash_2_dot_3_slash_slash_slash"

(org-roam--title-to-slug "10-2+3=45")
"10_2_3_45"
```

**Comment by jethrokuan** [comment]
Thanks @herbertjones . I have no objections to including accented characters in the filenames, I just didn't want the filenames to be invalid. Open to merging that in.

**Comment by chip2n** [comment]
The solution proposed by @herbertjones also solves the issue where `org-roam--title-to-slug "org-roam"` returns "orgroam" instead of the expected "org_roam".

**Comment by jethrokuan** [comment]
This is closed via #164
